src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs

```diff
@@ -79,7 +79,17 @@ public PreviewControl(NodeViewModel nodeViewModel)
         {
             this.scheduler = nodeViewModel.DynamoViewModel.Model.Scheduler;
             this.nodeViewModel = nodeViewModel;
-            InitializeComponent();
+            try
+            {
+                InitializeComponent();
+            }
+            catch(System.Windows.Markup.XamlParseException e)
+            {
+                // See https://docs.microsoft.com/en-us/dotnet/api/system.windows.markup.xamlparseexception?view=windowsdesktop-6.0#remarks
+                var args = new Core.CrashPromptArgs(e);
+                var ex = new Exception(args.Details);
+                Logging.Analytics.TrackException(ex, false);
```

Comment:
Hey @aparajit-pratap - I'm looking at this as well, and IMO - this will not be helpful - 

The stacktrace for this exception will be null, and more importantly, the `message` never makes it to analytics - see this code:

https://git.autodesk.com/Dynamo/Analytics.NET/blob/master/src/Core/events/ExceptionEvent.cs#L20

it only grabs stack trace and exception type.

I have two proposals that I will send PRs for.

1. create a new wrapper exception type in Dynamo that allows setting the stackTrace and type.
2. add a new TrackException API in Analytics Net that allows sending a string or series of strings directly.

Reply:
Thanks @mjkkirschner, in that case, I'll wait for your PR with the new analytics API or exception type, then update this PR accordingly.

Reply:
let me know what you think: 
https://github.com/DynamoDS/Dynamo/pull/12396

if this is too strange I can add investigate adding a new API to analytics.